src/clients/cassandra/CassandraConnector.js

```diff
@@ -1,24 +1,90 @@
 'use strict';
 
 const Promise = require('promise');
+const cassandra = require('cassandra-driver');
+const asyncEachLimit = require('async/eachLimit');
 
-/** Execute a batch of mutations
- * @param {Array<{mutation: string, params: Array<string|number|map}>} mutations
+/** @see http://docs.datastax.com/en/developer/nodejs-driver/3.2/features/batch/
+ * TODO: ASYNC_BATCH_LIMIT will need to be tuned.
+ * It is best to keep the batch size small (in the order of 10s).
+ * If the batch size is greater than 5k, then the server will issue a warning.
+ */
+const ASYNC_BATCH_LIMIT = process.env.ASYNC_BATCH_LIMIT || 10;
+
+/** Default Cassandra Client Options
+ * @see https://github.com/datastax/nodejs-driver/blob/master/lib/client-options.js
+ * 
+ * Cassandra Client Configurable Options
+ * @see https://docs.datastax.com/en/developer/nodejs-driver/3.2/api/type.ClientOptions/
+ * 
+ * Since datastax node driver does not have much detail on pooling, the java driver doc will be used:
+ * @see http://docs.datastax.com/en/developer/java-driver/3.3/manual/pooling/
+ * 
+ * Connection pools have a variable size, which gets adjusted automatically depending on the current load. 
+ * There will always be at least the core number of connections. The connection pool will pick 
+ * the connection with the minimum number of in-flight requests.
+ * 
+ * Cassandra version 2.1 or greater can send up to 32768 requests (stream ids) per connection.
+ * 
+ * CORE_CONNECTIONS_PER_HOST: to tune this option see java-driver 3.3 manual: 'Tuning protocol v3 for very high throughputs'
+ * 
+ * TODO: consider sslOptions and authProvider
+ */
+const distance = cassandra.types.distance;
+const CORE_CONNECTIONS_PER_HOST = 1;
+const options = {
+  contactPoints: [process.env.CASSANDRA_CONTACT_POINTS],
+  keyspace: process.env.CASSANDRA_KEYSPACE,
+  pooling: {
+    coreConnectionsPerHost: {
+      [distance.local]: CORE_CONNECTIONS_PER_HOST,
+      [distance.remote]: CORE_CONNECTIONS_PER_HOST
```

Comment:
No, I don't think so. For now I have the CORE_CONNECTIONS_PER_HOST, so I think I will instead add CORE_CONNECTIONS_PER_HOST_LOCAL and CORE_CONNECTIONS_PER_HOST_REMOTE